TestHelpers.Tests/MockFileTests.cs

```diff
@@ -290,6 +290,73 @@ public void MockFile_GetAttributeWithIllegalParameter_ShouldThrowOneArgumentExce
             Assert.Throws<ArgumentException>(action);
         }
 
+        [Test]
+        public void MockFile_SetAttributeOfExistingFile_ShouldReplaceExistingFlags()
+        {
+            var filedata = new MockFileData("test")
+            {
+                Attributes = FileAttributes.Normal
+            };
+            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
+            {
+                {XFS.Path(@"c:\something\demo.txt"), filedata},
+            });
+
+            fileSystem.File.SetAttributes(XFS.Path(@"c:\something\demo.txt"), FileAttributes.Hidden);
+            var attributes = fileSystem.File.GetAttributes(XFS.Path(@"c:\something\demo.txt"));
+            Assert.That(attributes, Is.EqualTo(FileAttributes.Hidden));
+        }
+
+        [Test]
+        public void MockFile_SetAttributeOfExistingDirectory_ShouldReplaceExistingFlags()
+        {
+            var fileSystem = new MockFileSystem();
+            var path = XFS.Path(@"c:\something");
+            fileSystem.AddDirectory(path);
+            var existingAttributes = fileSystem.File.GetAttributes(path);
+
+            fileSystem.File.SetAttributes(path, existingAttributes | FileAttributes.Hidden);
+            var attributes = fileSystem.File.GetAttributes(path);
+            Assert.That(attributes, Is.EqualTo(existingAttributes | FileAttributes.Hidden));
+        }
+
+        [Test]
+        [TestCase("", FileAttributes.Normal)]
+        [TestCase("   ", FileAttributes.Normal)]
+        [TestCase(">", FileAttributes.Normal)]
+        public void MockFile_SetAttributeOfExistingDirectoryWithInvalidParameters_ShouldThrowArgumentException(string path, FileAttributes attributes)
+        {
+            var fileSystem = new MockFileSystem();
+
+            TestDelegate action = () => fileSystem.File.SetAttributes(path, attributes);
+
+            Assert.Throws<ArgumentException>(action);
+        }
+
+        [Test]
+        public void MockFile_SetAttributeOfExistingDirectoryWithPathTooLong_ShouldThrowPathTooLongException()
+        {
+            var path = XFS.Path("C:\\8e95d05bc8d747d194e59516f8d1493b\\aa3a4391679642f9bf83f9501f7fb0e1\\8c51cab5b63543909fb7da59a3932b47\\49910ca19d0a4b59aadd4276829cc476\\3b0540cbd57b4770a52166c0ffd72e10\\70ab25de331e473a9cffbc5e297c79c1\\607c63a6f57645aa98073e9df7c7edd8\\dc473e93228c44c08d127c36d9cd9b01");
```

Comment:
Instead of this approach, I think calling something like string.PadRight(<some max number here>) would be a little nicer.